subprojects/core/src/main/java/org/gradle/internal/service/scopes/BuildScopeServices.java

```diff
@@ -287,8 +287,11 @@ public boolean isValid() {
             classLoaderHierarchyHasher);
     }
 
-    protected ScriptPluginFactory createScriptPluginFactory() {
-        return new ScriptPluginFactorySelector(defaultScriptPluginFactory(), this);
+    protected ScriptPluginFactory createScriptPluginFactory(BuildOperationExecutor buildOperationExecutor) {
+        DefaultScriptPluginFactory defaultScriptPluginFactory = defaultScriptPluginFactory();
+        ScriptPluginFactorySelector scriptPluginFactorySelector = new ScriptPluginFactorySelector(defaultScriptPluginFactory, this, buildOperationExecutor);
+        defaultScriptPluginFactory.setScriptPluginFactory(scriptPluginFactorySelector);
```

Comment:
Like this it is not possible, because we construct two objects that reference each other in both directions. Only one can have the other as constructor argument.